extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js

```diff
@@ -481,6 +505,96 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
     }
   }
 
+  /**
+   * Sends RTC request as specified by rtcConfig. Returns promise which
+   * resolves to the targeting informtation from the request response.
+   * If the publisher has specified that noCache == true, then we will only
+   * send one RTC request per page, and it will always bypass cache. If
+   * noCache !== true, then we always send two requests. The first will try
+   * to hit browser cache, and the second request will always bypass.
+   * This is a way for us to implement a simple stale-while-revalidate
+   * model.
+   * @return {?Promise} The rtc targeting info to attach to the ad url.
+   * @private
+   */
+  executeRtc_() {
+    if (rtcPromise) {
+      return this.mergeRtc();
+    }
+    const ampRtcPageElement = document.getElementById('amp-rtc');
+    if (!ampRtcPageElement) {
+      return Promise.resolve();
+    }
+    let endpoint;
+    const rtcConfig = tryParseJson(ampRtcPageElement.textContent);
+    if (!isObject(rtcConfig) || !(endpoint = rtcConfig['endpoint'])
+        || typeof endpoint != 'string' || !isSecureUrl(endpoint)) {
+      user().warn(TAG, 'invalid RTC config...');
+      return Promise.resolve();
+    }
+    this.shouldSendRequestWithoutRtc = err => {
+      return rtcConfig['sendAdRequestOnFailure'] !== false ?
```

Comment:
I'll make it check for 'false' as well, that makes sense to me. The docs specify what to put though, so if someone puts 0 or '0' I think that's on them, no? I'm not even sure how '0' should be interpreted.